### PR TITLE
Assorted linting fixes.

### PR DESCRIPTION
**Pull request by markusthoemmes**
There's a lot of linter juice in knative.dev/pkg. This fixes almost all of the warnings except for capitalized error strings.

### CODE REVIEW

File: injection/informers_test.go [hunk 1]
```diff
@@ -61,8 +61,8 @@ func TestRegisterInformersAndSetup(t *testing.T) {
 	i.RegisterInformer(injectFooInformer)
 	i.RegisterInformer(injectBarInformer)
 
-	ctx, infs := context.Background(), []controller.Informer{}
-	ctx, infs = i.SetupInformers(ctx, &rest.Config{})
+	ctx, _ := context.Background(), []controller.Informer{}
```
Comment (vagababov) [inline injection/informers_test.go]: This line looks incorrect.

Comment (markusthoemmes) [reply injection/informers_test.go]: Indeed, that's what you get for being braindead about a bunch of linter output.

File: apis/duck/cached_test.go [hunk 2]
```diff
@@ -39,16 +38,12 @@ var _ InformerFactory = (*BlockingInformerFactory)(nil)
 func (bif *BlockingInformerFactory) Get(gvr schema.GroupVersionResource) (cache.SharedIndexInformer, cache.GenericLister, error) {
 	atomic.AddInt32(&bif.getting, 1)
 	// Wait here until we can acquire the lock!
-	bif.lock.Lock()
-	bif.lock.Unlock()
+	<-bif.block
```
Comment (evankanderson) [inline apis/duck/cached_test.go]: Should this comment be changed to "until the channel is closed"?

**Pull request merged**